### PR TITLE
Migration to RouterProvider #235

### LINKED ISSUE #235: use RouterProvider react  router v6
https://reactrouter.com/en/main/upgrading/v6-data

### PR DESCRIPTION
**Pull request by MatteoGuarnaccia5**
## Description

Migrated routes to use RouterProvider v6. refactored tests to use new logic.

## Testing instructions

- [ ] Review code
- [ ] Check Actions build
- [ ] Review changes to test coverage

## Agile board tracking

closes #235

### CODE REVIEW

File: src/view/viewTabs.component.tsx [hunk 1]
```diff
@@ -135,12 +110,12 @@ function ViewTabs() {
...
           >
             <TabPanel value={value} label={value}>
-              {routing}
+              <Outlet />
             </TabPanel>
           </Box>
         </>
       ) : (
-        routing
+        <></>
```
Comment (joshuadkitenge) [inline src/view/viewTabs.component.tsx:118]:
```suggestion
        <Outlet />
```

File: src/view/viewTabs.component.tsx [hunk 2]
```diff
@@ -3,19 +3,12 @@ import Tab from '@mui/material/Tab';
...
-import Items from '../items/items.component';
-import ItemsLandingPage from '../items/itemsLandingPage.component';
-import Manufacturer from '../manufacturer/manufacturer.component';
-import ManufacturerLandingPage from '../manufacturer/manufacturerLandingPage.component';
-import Systems from '../systems/systems.component';
 import { getSciGatewayPageHeightCalc, isRunningInDevelopment } from '../utils';
-import { HomePage } from '../homePage/homePage.component';
 
 export const paths = {
+  any: '*',
```
Comment (joelvdavies) [inline src/view/viewTabs.component.tsx:11]: Could this also be moved to `App.tsx` and then be used inside the router itself.

File: src/App.tsx [hunk 3]
```diff
@@ -69,28 +77,60 @@ const App: React.FunctionComponent = () => {
...
   }, []);
 
+  return <div className="Root"></div>;
+};
+
+const router = createBrowserRouter([
+  {
+    path: '/*',
+    Component: Layout,
+    children: [
```
Comment (joelvdavies) [inline src/App.tsx:87]: Should this even have a path? The migration guide FAQ suggests it should be pathless.

File: src/App.tsx [hunk 4]
```diff
@@ -47,7 +55,7 @@ const queryClient = new QueryClient({
   },
 });
 
-const App: React.FunctionComponent = () => {
+const Root: React.FunctionComponent = () => {
```
Comment (joelvdavies) [inline src/App.tsx:58]: I don't think `Root` is actually needed any more after the migration - I would assume Layout would deal with rendering something. The contents of this though should probably be in `Layout` as right now none of it does anything.

**Review by joelvdavies — CHANGES_REQUESTED** [review]
Haven't looked at the test changes yet, but I did notice an issue with the plugin refresh not doing anything.

Also just to raise, I saw https://reactrouter.com/en/main/start/tutorial#handling-not-found-errors allows us to have 404 pages for invalid ids, and more general error pages we could look at implementing.

File: src/view/breadcrumbs.component.test.tsx [hunk 5]
```diff
@@ -1,15 +1,19 @@
...
 describe('Channel Breadcrumbs', () => {
   const onChangeNode = vi.fn();
   const onChangeNavigateHome = vi.fn();
   let props: BreadcrumbsProps;
 
   const createView = () => {
-    return renderComponentWithBrowserRouter(<Breadcrumbs {...props} />);
+    return renderComponentWithRouterProvider(
+      <Breadcrumbs {...props} />,
+      paths.root
```
Comment (joelvdavies) [inline src/view/breadcrumbs.component.test.tsx:15]:
```suggestion
```
I don't think this does anything in the most recent commit.

File: src/testUtils.tsx [hunk 6]
```diff
@@ -30,56 +35,34 @@ interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
   queryClient?: QueryClient;
 }
 
-export function renderComponentWithBrowserRouter(
+export function renderComponentWithRouterProvider(
   ui: React.ReactElement,
+  urlPathKey?: keyof typeof paths,
+  pathName?: string,
```
Comment (joelvdavies) [inline src/testUtils.tsx:41]:
```suggestion
  initialEntry?: string,
```
I think this makes it a bit clearer.

File: src/view/viewTabs.component.test.tsx [hunk 7]
```diff
@@ -20,13 +20,7 @@ describe('View Tabs', () => {
     expect(viewTabs.getByRole('tab', { selected: true })).toHaveTextContent(
       'Catalogue'
     );
-    const catalogueHomeButton = await screen.findByRole('button', {
-      name: 'navigate to catalogue home',
-    });
-    expect(catalogueHomeButton).toBeInTheDocument();
-    await waitFor(() => {
-      expect(screen.getByText('Beam Characterization')).toBeInTheDocument();
-    });
```
Comment (joelvdavies) [inline src/view/viewTabs.component.test.tsx:29]: Why was this removed? It ensures the routing renders the appropriate component. Same with the other removals below.

**Review by joelvdavies — COMMENTED** [review]
A couple comments from the tests.

File: src/view/viewTabs.component.tsx (hunk 2 above)
Comment (joelvdavies) [reply src/view/viewTabs.component.tsx:11]: @MatteoGuarnaccia5 I think you missed out this.

Comment (MatteoGuarnaccia5) [reply src/view/viewTabs.component.tsx:11]: I did add it to `App.tsx`, this any was just for `vewTabs.component.test.tsx` as it doesn't work with any other of the paths

Comment (joelvdavies) [reply src/view/viewTabs.component.tsx:11]: Ah sorry, I meant the whole `paths` object so they are imported from App.tsx and used inside App.tsx inside the router for the paths.

**Review by joelvdavies — APPROVED** [review]
Tested locally in dev, locally in SciGateway with and without auth (checking token expiry works) and with the dev docker compose used on the dev machine.

**Review by joshuadkitenge — APPROVED** [review]
